### PR TITLE
Add snapshot compression option

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

bz2 is horribly horribly slow

#### Summary of Changes

Add gzip and zstd as options for snapshot compression

Fixes #

### CODE REVIEW

**Comment by mvines** [comment]
FYI:
https://github.com/solana-labs/solana/blob/4febf8a947b4dfbf1e3e3646995ad0a6e25b1d41/core/src/rpc_service.rs#L46

**Comment by sakridge** [comment]
> 
> 
> FYI:
> 
> https://github.com/solana-labs/solana/blob/4febf8a947b4dfbf1e3e3646995ad0a6e25b1d41/core/src/rpc_service.rs#L46

Thanks. I got that one.

File: validator/src/main.rs [hunk 1]
```diff
@@ -838,6 +848,16 @@ pub fn main() {
...
 
+    let mut snapshot_compression = CompressionType::Bzip2;
+    if let Ok(compression_str) = value_t!(matches, "snapshot_compression", String) {
+        match compression_str.as_str() {
+            "bz2" => snapshot_compression = CompressionType::Bzip2,
+            "gzip" => snapshot_compression = CompressionType::Gzip,
+            "zstd" => snapshot_compression = CompressionType::Zstd,
+            "none" => snapshot_compression = CompressionType::NoCompression,
+            _ => panic!("Compression type not recognized: {}", compression_str),
+        }
```
Comment (mvines) [inline validator/src/main.rs:859]: Use https://docs.rs/clap/2.33.0/clap/struct.Arg.html#method.possible_values to remove the panic! here.  User input should never cause a panic

Comment (sakridge) [reply validator/src/main.rs:860]: I added the possible_values so this should never be hit.

**Pull request merged**